### PR TITLE
Work towards scalachecking the resolver API.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Command.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.genivi.sota.resolver.packages.Package
+import org.genivi.sota.resolver.test.{Result, Success, Failure, SuccessPackage}
+import org.genivi.sota.resolver.test.{VehicleRequestsHttp, PackageRequestsHttp, FilterRequestsHttp}
+import org.genivi.sota.resolver.vehicles.Vehicle
+import org.scalacheck.{Arbitrary, Gen}
+import scala.annotation.tailrec
+import Misc._
+
+
+abstract sealed trait Command
```
Comment (koshelev) [inline external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Command.scala]: just 'trait' would be enough, traits are always abstract

**Comment by stevana** [comment]
Fixed comments.

**Comment by stevana** [comment]
Merge.

**Comment by stevana** [comment]
@jerrytrieu: @koshelev said I should ask you to merge this. It only touches the tests.

**Pull request merged**